extensions/vscode/src/extension.ts

```diff
@@ -1,41 +1,49 @@
-'use strict';
-
-import * as net from 'net';
-
-import {Trace} from 'vscode-jsonrpc';
-import { window, workspace, commands, ExtensionContext, Uri } from 'vscode';
-import { LanguageClient, LanguageClientOptions, StreamInfo, Position as LSPosition, Location as LSLocation } from 'vscode-languageclient';
-
-export function activate(context: ExtensionContext) {
-    // The server is a started as a separate app and listens on port 5007
-    let connectionInfo = {
-        port: 5007
-    };
-    let serverOptions = () => {
-        // Connect to language server via socket
-        let socket = net.connect(connectionInfo);
-        let result: StreamInfo = {
-            writer: socket,
-            reader: socket
-        };
-        return Promise.resolve(result);
-    };
-    
-    let clientOptions: LanguageClientOptions = {
-        documentSelector: ['Solidity'],
-        synchronize: {
-            fileEvents: workspace.createFileSystemWatcher('**/*.*')
-        }
-    };
-    
-    // Create the language client and start the client.
-    let lc = new LanguageClient('Xtext Server', serverOptions, clientOptions);
-
-    // enable tracing (.Off, .Messages, Verbose)
-    lc.trace = Trace.Verbose;
-    let disposable = lc.start();
-    
-    // Push the disposable to the context's subscriptions so that the 
-    // client can be deactivated on extension deactivation
-    context.subscriptions.push(disposable);
+'use strict';
+
+import * as path from 'path';
+import * as os from 'os';
+
+import {Trace} from 'vscode-jsonrpc';
+import { commands, window, workspace, ExtensionContext, Uri } from 'vscode';
+import { LanguageClient, LanguageClientOptions, ServerOptions } from 'vscode-languageclient';
+
+export function activate(context: ExtensionContext) {
+    // The server is a locally installed in src/mydsl
+    let launcher = os.platform() === 'win32' ? 'solidity-ls.bat' : 'solidity-ls';
+    let script = context.asAbsolutePath(path.join('src', 'ls', 'bin', launcher));
+    
+    let serverOptions: ServerOptions = {
+        run : { command: script },
+        debug: { command: script, args: ['-Xdebug','-Xrunjdwp:server=y,transport=dt_socket,address=8000,suspend=n,quiet=y','-Xmx256m'] }
+    };
+    
+    let clientOptions: LanguageClientOptions = {
+        documentSelector: ['Solidity'],
+        synchronize: {
+            fileEvents: workspace.createFileSystemWatcher('**/*.*')
+        }
+    };
+    
+    // Create the language client and start the client.
+    let lc = new LanguageClient('Solidity Language Server', serverOptions, clientOptions);
+    
+    var disposable2 =commands.registerCommand("sol.a.proxy", async () => {

```

Comment:
these commands make no sense. we should remove this down to context.subscriptions.push(disposable2);